libnmstate/state.py

```diff
@@ -768,6 +769,9 @@ def merge_route_rules(self, other_state):
 def dict_update(origin_data, to_merge_data):
     """Recursevely performes a dict update (merge)"""
 
+    if not to_merge_data:
```

Comment:
shouldn't this be `if to_merge_data == {} ` instead ?

Reply:
`not to_merge_data` includes this case and more.
As it can only be an instance of `Mapping`, we should be fine with the generic version.